packages/@expo/cli/src/run/remoteBuildCache.ts

```diff
@@ -168,3 +175,24 @@ function importFingerprintForDev(projectRoot: string): null | typeof import('@ex
     return null;
   }
 }
+
+function isDevClientBuild({
+  runOptions,
+  projectRoot,
+}: {
+  runOptions: AndroidRunOptions | IosRunOptions;
+  projectRoot: string;
+}) {
+  if (!hasDirectDevClientDependency(projectRoot)) {
+    return false;
```

Comment:
what will happen if i run `npx run:ios` (debug build) but without dev-client installed? will it download a debug build or release build?

Reply:
Hmm, that's a great question, it would download the latest build that was uploaded and wouldn't check for debug build or release build. I don't think we currently have a way to difenciate these builds on EAS

Reply:
hopefully we could resolve this problem in some ways. to download undetermined build type is not ideal. maybe we put more information in build metadata like what we did for developmentClient

Reply:
thank differently. maybe i can add a custom metadata in fingerprint and allows you to pass build type or build variants. does that feel easier?

Reply:
I think there is some value in having this type of metadata in EAS but this wouldn't really solve the issues for other provides, so I would say that having some sort of custom metadata in fingerprint would be quite useful

Reply:
sg! i'll plan for fingerprint lib side. maybe you need further update `eas fingerprint:generate` to pass the custom metadata then.